simulation/tools/sim_viewer.cpp

```diff
@@ -36,66 +36,64 @@
  * $Id: pcd_viewer.cpp 5094 2012-03-15 01:03:51Z rusu $
  *
  */
+#include <boost/shared_ptr.hpp>
+
 #include <Eigen/Dense>
 #include <Eigen/Geometry>
-#include <boost/shared_ptr.hpp>
+
 #include <cmath>
 #include <iostream>
 #ifdef _WIN32
 #define WIN32_LEAN_AND_MEAN
 #include <windows.h>
 #endif
 
-#include <GL/glew.h>
-
 #include <pcl/pcl_config.h>
+
+#include <GL/glew.h>
 #ifdef OPENGL_IS_A_FRAMEWORK
 #include <OpenGL/gl.h>
 #else
 #include <GL/gl.h>
 #endif
 
+#include <pcl/features/normal_3d.h>
 #include <pcl/io/pcd_io.h>
 #include <pcl/point_types.h>
+#include <pcl/surface/gp3.h>
 
 #include "pcl/common/common.h"
 #include "pcl/common/transforms.h"
 
-#include <pcl/features/normal_3d.h>
-#include <pcl/io/pcd_io.h>
-#include <pcl/point_types.h>
-#include <pcl/surface/gp3.h>
-
 // define the following in order to eliminate the deprecated headers warning
 #define VTK_EXCLUDE_STRSTREAM_HEADERS
+#include <pcl/console/parse.h>
+#include <pcl/console/print.h>
+#include <pcl/console/time.h>
 #include <pcl/io/vtk_lib_io.h>
 
 #include "pcl/simulation/camera.h"
 #include "pcl/simulation/model.h"
 #include "pcl/simulation/range_likelihood.h"
 #include "pcl/simulation/scene.h"
 
-#include <pcl/console/parse.h>
-#include <pcl/console/print.h>
-#include <pcl/console/time.h>
-
 // RangeImage:
 #include <pcl/range_image/range_image_planar.h>
 
 // Pop-up viewer
-#include <pcl/visualization/cloud_viewer.h>
-
-#include <cfloat>
 #include <pcl/common/common.h>
 #include <pcl/console/parse.h>
 #include <pcl/console/print.h>
 #include <pcl/console/time.h>
 #include <pcl/io/pcd_io.h>
+#include <pcl/visualization/cloud_viewer.h>
 #include <pcl/visualization/histogram_visualizer.h>
 #include <pcl/visualization/keyboard_event.h>
 #include <pcl/visualization/pcl_visualizer.h>
 #include <pcl/visualization/point_cloud_handlers.h>
 #include <pcl/visualization/point_picking_event.h>
```

Comment:
I meant those lines will be affected as this regex pattern doesn't take into account the quotes includes.
The pcl includes are not "together".

Reply:
As I mentioned in initial post: If there is a macro (e.g. `#if`, `#define`), clang-format will handle the includes before and after this macros separately, as it doesn't know if this macro has any effect on these includes. I will check it, wenn the include order is accepted. Until than adjusted header and source files are just to see the result without manual adjustments ;-)